internal/util/diff/pkgdiff_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"testing"
+
+	"github.com/GoogleContainerTools/kpt/internal/testutil/pkgbuilder"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/sets"
+)
+
+func TestPkgDiff(t *testing.T) {
+	testCases := []struct {
+		name string
+		pkg1 *pkgbuilder.RootPkg
+		pkg2 *pkgbuilder.RootPkg
+		diff sets.String
+	}{
+		{
+			name: "equal packages doesn't have a diff",
+			pkg1: pkgbuilder.NewRootPkg().
+				WithKptfile(pkgbuilder.NewKptfile().
+					WithSetters(
+						pkgbuilder.NewSetter("foo", "bar"),
+					),
+				).
+				WithResource(pkgbuilder.DeploymentResource),
+			pkg2: pkgbuilder.NewRootPkg().
+				WithKptfile(pkgbuilder.NewKptfile().
+					WithSetters(
+						pkgbuilder.NewSetter("foo", "bar"),
+					),
+				).
+				WithResource(pkgbuilder.DeploymentResource),
+			diff: toStringSet(),
+		},
+		{
+			name: "different files between packages",
+			pkg1: pkgbuilder.NewRootPkg().
+				WithKptfile().
+				WithResource(pkgbuilder.DeploymentResource),
+			pkg2: pkgbuilder.NewRootPkg().
+				WithKptfile().
+				WithResource(pkgbuilder.ConfigMapResource),
+			diff: toStringSet("configmap.yaml", "deployment.yaml"),
+		},
+		{
+			name: "different setters in Kptfile is a diff",
+			pkg1: pkgbuilder.NewRootPkg().
```

Comment:
Does order of setter definitions in Kptfile generate diff?

Reply:
Good point. Ordering (and formatting) is a challenge for us when we do diffs. We should probably aim to have one implementation of diff, and then we can consider whether we want to implement solutions here that are more resilient.